plugins/catalog-backend/src/service/CatalogBuilder.ts

```diff
@@ -417,6 +419,15 @@ export class CatalogBuilder {
     return this;
   }
 
+  /**
+   * Enables the publishing of events for conflicting entities so that a subscriber
+   * can listen to them and implement a conflict resolution strategy like retries.
+   */
+  setConflictEventBroker(broker: EventBroker): CatalogBuilder {
```

Comment:
There seems to be a pattern here using setters rather than expanding the catalog env. I was told to use the setters for a previous PR for this reason so went with that here too.

Reply:
There is another pull request in the works that passes the eventBroker in the env. However that is not to say that is the right thing.

https://github.com/backstage/backstage/pull/16678

However what you have done here will make the event integration more intentional for the catalog, which might not be a bad thing.. basically I am not sure whether this should be intenional or "just work" for backstage's that have the eventBroker set in the environment.